assets/js/i18n/locales.js

```diff
@@ -16,11 +18,22 @@ function isSupported (locale) {
 }
 
 /**
- * Reload the page, adding a "locale" query parameter to the URL
+ * Ask the server to change the current locale and go to the page instructed
  */
-export function setLocale (locale) {
-  // very quick and dirty solution for reloading with added parameter
-  const q = new URLSearchParams(window.location.search)
-  q.set('locale', locale)
-  window.location.search = '?' + q.toString()
+export async function setLocale (locale) {
```

Comment:
Please change this logic to create an HTTP `<form>` and submit it. Perhaps even make locale-switcher a React component that renders a `<form>` (and an accompanying Django template, for Django-allauth pages).

(Rationale: this function as written melds `fetch`, HTTP headers, HTTP 302 Found, JSON, `window.location` and Promises. That's too many concepts.)

Reply:
With your permission, I'll build the form in js for now and I'll consider moving the form to a React component when we have the design for the switcher.

Reply:
Why?

Reply:
So that I don't change the React component twice

Reply:
Uh ... I'm not seeing it.

Here's what I'm suggesting: `LocaleSwitcher` gets a `<form>` with `csrfToken`; `next` (populated by `window.location.href`);and `locale` (the selector you have now). The only event handler: `onChangeLocale` calls `form.submit()`.

Then we nix `setLocale()` forever.

I feel like that's less work than building a new `setLocale()` function. What am I missing?